core/src/main/java/jenkins/security/UpdateSiteWarningsMonitor.java

```diff
@@ -0,0 +1,176 @@
+ * - Intersection of active and inapplicable
+ *
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class UpdateSiteWarningsMonitor extends AdministrativeMonitor {
+    @Override
+    public boolean isActivated() {
+        return !getActiveCoreWarnings().isEmpty() || !getActivePluginWarningsByPlugin().isEmpty();
+    }
+
+    public List<UpdateSite.Warning> getActiveCoreWarnings() {
+        List<UpdateSite.Warning> CoreWarnings = new ArrayList<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!UpdateSite.Warning.TYPE_CORE.equals(warning.type)
+                    || !UpdateSite.Warning.NAME_CORE.equals(warning.name)) {
+                // this is not a core warning
+                continue;
+            }
+            CoreWarnings.add(warning);
+        }
+        return CoreWarnings;
+    }
+
+    public Map<PluginWrapper, List<UpdateSite.Warning>> getActivePluginWarningsByPlugin() {
+        Map<PluginWrapper, List<UpdateSite.Warning>> activePluginWarningsByPlugin = new HashMap<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!UpdateSite.Warning.TYPE_PLUGIN.equals(warning.type)) {
+                // this is not a plugin warning
+                continue;
+            }
+
+            String pluginName = warning.name;
+
+            PluginWrapper plugin = Jenkins.getInstance().getPluginManager().getPlugin(pluginName);
+
+            if (!activePluginWarningsByPlugin.containsKey(plugin)) {
+                activePluginWarningsByPlugin.put(plugin, new ArrayList<UpdateSite.Warning>());
+            }
+            activePluginWarningsByPlugin.get(plugin).add(warning);
+        }
+        return activePluginWarningsByPlugin;
+
+    }
+
+    private Set<UpdateSite.Warning> getActiveWarnings() {
+        ExtensionList<UpdateSiteWarningsConfiguration> configurations = ExtensionList.lookup(UpdateSiteWarningsConfiguration.class);
+        if (configurations.isEmpty()) {
+            return Collections.emptySet();
+        }
+        UpdateSiteWarningsConfiguration configuration = configurations.get(0);
+
+        HashSet<UpdateSite.Warning> warnings = new HashSet<>(configuration.getApplicableWarnings());
+
+        for (Iterator<UpdateSite.Warning> it = warnings.iterator(); it.hasNext(); ) {
+            UpdateSite.Warning warning = it.next();
+
+            if (configuration.getIgnoredWarnings().contains(warning.id)) {
+                it.remove();
+            }
+        }
+
+        return Collections.unmodifiableSet(warnings);
+    }
+
+    /**
+     * Redirects the user to the security configuration where they can configure which warnings to show/hide.
+     *
+     * For Stapler use only.
+     *
+     * @param req the request
+     * @param rsp the response
+     * @throws IOException
+     */
+    public void doAct(StaplerRequest req, StaplerResponse rsp, @QueryParameter String fix, @QueryParameter String configure) throws IOException {
```

Comment:
Delete the `req` and `rsp` parameters and return `HttpResponses.redirectViaContextPath` instead.